liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/server/DevMojo.java

```diff
@@ -366,10 +368,20 @@ protected void doExecute() throws Exception {
                         }
                     } else if (directory.startsWith(configDirectory.toPath())) {  // config files                                                                    
                         if (fileChanged.exists() && (event.kind() == StandardWatchEventKinds.ENTRY_MODIFY ||  event.kind() == StandardWatchEventKinds.ENTRY_CREATE)) {
-                            copyFile(fileChanged, configDirectory, serverDirectory);
+                            // config directory does not exist, refresh only config file
+                            if (noConfigDir && fileChanged.getAbsolutePath().endsWith(configFile.getName())){ 
+                                copyFile(fileChanged, configDirectory, serverDirectory);
+                            } else if (!noConfigDir) { // config directory exists, refresh entire directory
+                                copyFile(fileChanged, configDirectory, serverDirectory);
+                            }
                         } else if (event.kind() == StandardWatchEventKinds.ENTRY_DELETE){
-                            log.debug("Config file deleted: " + fileChanged.getName());
-                            deleteFile(fileChanged, configDirectory, serverDirectory);
+                            if (noConfigDir && fileChanged.getAbsolutePath().endsWith(configFile.getName())){ 
```

Comment:
Same as above, combine if and else if statements to
`!noConfigDir || fileChanged.getAbsolutePath().endsWith(configFile.getName())`